### PR TITLE
Add __all__ to __init__.py files

### LINKED ISSUE #420: Let IDE(s) know public symbols
## 🚀 Feature (or Bug?)
I'm using Pylance on VSCode, and it doesn't autocomplete, for example, `Accuracy` even if I typed the following:

```python
from torchmetrics import Ac
```

This is because imported symbols are considered private by default (by at least pylance (pyright) as documented [here](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface)).

### Motivation
For better coding experience (autocomplete + docstring reference) :]

### Pitch
As stated in https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface, we have the following options to explicitly make symbols public.

#### 1. Add `__all__` to `__init__.py` files.
```python
# torchmetrics/__init__.py
from torchmetrics.classification import (
    Accuracy,
    ...
)
__all__ = [
    "Accuracy",
    ...
]
```

#### 2. Use wildcard imports.
```python
# torchmetrics/__init__.py
from torchmetrics.classification import *

# torchmetrics/classification/__init__.py
__all__ = [
    "Accuracy",
    ...
]
```

#### 3. Use redundant symbol aliases.
```python
# torchmetrics/__init__.py
from torchmetrics.classification import (
    Accuracy as Accuracy,
    ...
)
```

### Additional context
https://github.com/microsoft/pylance-release/issues/1032
https://github.com/microsoft/pyright/pull/1877

### PR DESCRIPTION
**Pull request by akihironitta**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [n/a] Did you make sure to update the docs?
- [n/a] Did you write any new necessary tests?

## What does this PR do?

Fixes #420.

### Description of the changes
- Add `__all__` only to user-facing `__init__.py` files (i.e. `torchmetrics/__init__.py` and `torchmetrics/functional/__init__.py`)
- Remove `# noqa: F401` as not needed anymore

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/__init__.py [hunk 1]
```diff
@@ -11,9 +11,10 @@
 _PACKAGE_ROOT = os.path.dirname(__file__)
 _PROJECT_ROOT = os.path.dirname(_PACKAGE_ROOT)
 
-from torchmetrics.audio import PIT, SI_SDR, SI_SNR, SNR  # noqa: E402, F401
-from torchmetrics.average import AverageMeter  # noqa: E402, F401
-from torchmetrics.classification import (  # noqa: E402, F401
+from torchmetrics import functional  # noqa: E402
```
Comment (akihironitta) [inline torchmetrics/__init__.py:14]: Without importing `functional` submodule here, pylance doesn't autocomplete:
```python
import torchmetrics
torchmetrics.func  # it doesn't autocomplete
```

**Review by Borda — APPROVED** [review]
Nice!

**Review by SkafteNicki — APPROVED** [review]
